docs/smart-contract-backend/scb.md

```diff
@@ -69,6 +69,35 @@ The node client is a process that talks to a Cardano node. It can submit transac
 2. Ask for all unspent outputs at an address (see 1.2.1.5)
 3. Ask for the current slot number (see 1.2.1.1)
 
+We communicate with the node client using a REST API, and the listener will establish a web sockets connection and be updated regularly with accepted transactions.
+
+The mock will implement the following endpoints (which will also be exposed by the real node):
+
+```
+--| Notations
+    <-   Client sends to Server.
+    ->   Server sends to Client.
+    <d>  Datum `d` is JSON encoded.
+
+-- API
+GET /utxo/:addr
+-> <[(TxOutRef, TxOut)]>
+
+GET /currentSlot
+-> <Slot>
+
+GET /chain/:slot -- Web sockets stream
+-> <[Tx]>
+(Possible error: `Specified slot outside of K range`)
+
+POST /tx
+<- <Tx>
+(Lots of possible errors during transaction validation)
```

Comment:
We could validate some of the properties of the transaction here (is it balanced, do the scripts pass), but realistically if this is the real node client we won't get any immediate feedback when submitting the tx.

Reply:
I think this will serve as both the chain and node client mocks.